src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs

```diff
@@ -94,16 +94,27 @@ private JSSynchronizationContext (Thread mainThread)
             }
         }
 
-        internal static void InstallSynchronizationContext (bool actuallyInstallForCurrentThread) {
-            // HACK: This exists for linker reasons
-            if (!actuallyInstallForCurrentThread)
+        [MethodImpl(MethodImplOptions.NoInlining)] // https://github.com/dotnet/runtime/issues/71425
+        // the marshaled signature is:
+        // void InstallSynchronizationContext()
+        internal static void InstallSynchronizationContext (JSMarshalerArgument* arguments_buffer) {
```

Comment:
if would be good to move this to `JavaScriptExports.cs` so that we have them all in the same place. That class is also already protected from trimming.

Reply:
I mean, the marshaling part of this method, the actual init, could stay here.